src/ctrl/em_network_topo.cpp

```diff
@@ -88,6 +94,90 @@ em_network_topo_t *em_network_topo_t::find_topology_by_bh_associated(mac_address
 	return NULL;
 }
 
+em_network_topo_t *em_network_topo_t::find_topology_by_bh_associated(dm_easy_mesh_t *dm)
+{
+	mac_addr_str_t bss_mac_str, dev_mac_str;
+
+	if (dm == NULL)	{
+		printf("%s:%d: dm is NULL, return NULL\n", __func__, __LINE__);
+		return NULL;
+	}
+
+	dm_easy_mesh_t::macbytes_to_string(dm->m_device.m_device_info.intf.mac, dev_mac_str);
+	// Obtain the mesh_sta information of this dm.
+	em_bss_info_t *bss = dm->get_bsta_bss_info();
+	if (bss == NULL) {
+		printf("%s:%d: Backhaul sta bss not found for dev_mac:%s, return NULL\n", __func__, __LINE__, dev_mac_str);
+		dm->m_device.m_device_info.backhaul_mac.media = em_media_type_ieee8023ab;
+		return NULL;
+	}
+	// From the bss info obtained, find the appropriate em_network_topo_t containing this bss
+	// and return that topology object.
+	mac_address_t bss_mac;
+	memcpy(bss_mac, bss->id.bssid, sizeof(mac_address_t));
+	dm_easy_mesh_t::macbytes_to_string(bss_mac, bss_mac_str);
+	if (dm->get_colocated() == false) {
+		// Update the backhaul of the dm object with the bss mac address
+		memcpy(dm->m_device.m_device_info.backhaul_mac.mac, bss_mac, sizeof(mac_address_t));
+		dm->m_device.m_device_info.backhaul_mac.media = em_media_type_ieee80211ac_5;
+		// check if backhaul mac is 00:00:00:00:00:00, then change the media type to em_media_type_ieee8023ab
+		if (memcmp(bss_mac, ZERO_MAC_ADDR, sizeof(mac_address_t)) == 0) {
+			dm->m_device.m_device_info.backhaul_mac.media = em_media_type_ieee8023ab;
+		}
+	}
+	printf("%s:%d: Find topology associated with bss mac: %s\n", __func__, __LINE__, bss_mac_str);
+	if (memcmp(bss_mac, ZERO_MAC_ADDR, sizeof(mac_address_t)) == 0) {
+		printf("%s:%d: bss mac is zero, return NULL\n", __func__, __LINE__);
+		return NULL;
+	}
+	// Now search for the topology containing this bss mac address.
+	return g_network_topology->find_topology_by_bss_mac(bss_mac);
+}
+
+em_network_topo_t *em_network_topo_t::find_topology_by_bss_mac(mac_address_t bss_mac)
+{
+	unsigned int i;
+	mac_addr_str_t bss_mac_str;
+	em_network_topo_t *topo;
+
+	dm_easy_mesh_t::macbytes_to_string(bss_mac, bss_mac_str);
+	for (i = 0; i < m_data_model->m_num_bss; i++) {
+		if (memcmp(m_data_model->m_bss[i].m_bss_info.id.bssid, bss_mac, sizeof(mac_address_t)) == 0) {
+			printf("%s:%d: Found topology of bss mac: %s num_child_topo:%d\n", __func__, __LINE__,
+				   bss_mac_str, this->m_num_topologies);
+			return this;
+		}
+	}
```

Comment:
I think this can be replaced by:

```suggestion
	em_bss_info_t* bss_info = m_data_model->get_bss_info_with_mac(bss_mac);
    EM_ASSERT_NULL(bss_info, this, "Found topology of bss mac: %s num_child_topo:%d", bss_mac_str, this->m_num_topologies);
```

`EM_ASSERT_NULL` will automatically expand and check if `bss_info` is NULL and if it is (meaning in this case it was found) then it will return `this`. The `__func__` and `__LINE__` are automatically used by the `em_printfout` macro which is called in the `EM_ASSERT_NULL`.

It's equivalent to:
```
if (bss_info == NULL) {
   em_printfout("Found topology of bss mac: %s num_child_topo:%d", bss_mac_str, this->m_num_topologies);
   return this;
}